internal/telemetry/telemetry.go

```diff
@@ -65,22 +63,24 @@ func (c *client) configChange(configuration []Configuration) {
 	}
 }
 
-// productEnabled enqueues an app-product-change event that signals a product has been turned on.
+// productChange enqueues an app-product-change event that signals a product has been `enabled`.
 // Must be called with c.mu locked. An app-product-change event with enabled=true indicates
 // that a certain product has been used for this application.
-func (c *client) productEnabled(namespace Namespace) {
+func (c *client) productChange(namespace Namespace, enabled bool) {
 	if !c.started {
 		log("attempted to send product change event, but telemetry client has not started")
 		return
 	}
-	products := new(Products)
+	products := new(ProductsPayload)
```

Comment:
Apparently, the payloads sent for this particular message were invalid, at least according to the telemetry API documentation... A `products` top-level key is required...